evm/src/cpu/kernel/asm/core/jumpdest_analysis.asm

```diff
@@ -42,7 +42,7 @@ continue:
 proof_ok:
     // stack: i, ctx, final_pos, retdest
     // We already know final_pos is a jumpdest
-    %stack (i, ctx, final_pos) -> (ctx, @SEGMENT_JUMPDEST_BITS, i)
+    %stack (i, ctx, final_pos) -> (ctx, @SEGMENT_JUMPDEST_BITS, final_pos)
```

Comment:
`i == final_pos` (this is the condition to enter), and using final_pos removes 1 SWAP per loop iteration